nsqdserver/http.go

```diff
@@ -563,6 +565,39 @@ func (s *httpServer) doDeleteChannel(w http.ResponseWriter, req *http.Request, p
 	return nil, nil
 }
 
+func (s *httpServer) doSkipChannel(w http.ResponseWriter, req *http.Request, ps httprouter.Params) (interface{}, error) {
+	_, topic, channelName, err := s.getExistingTopicChannelFromQuery(req)
+	if err != nil {
+		return nil, err
+	}
+
+	channel, err := topic.GetExistingChannel(channelName)
+	if err != nil {
+		return nil, http_api.Err{404, "CHANNEL_NOT_FOUND"}
+	}
+
+	nsqd.NsqLogger().Logf("topic:%v channel:%v ", topic.GetTopicName(), channel.GetName())
+
+	if strings.Contains(req.URL.Path, "unskip") {
+		//update channel state before set channel consume offset
+		err = s.ctx.UpdateChannelState(channel, -1, 0)
+		err = s.ctx.nsqdCoord.SetChannelConsumeOffsetToCluster(channel, int64(channel.GetConfirmed().Offset()), channel.GetConfirmed().TotalMsgCnt(), true)
+		if err != nil {
+			nsqd.NsqLogger().Errorf("fail to set channel consume offset channel %v topic %v, unskip quit.", channel.GetName(), topic.GetTopicName())
+		}
+	} else {
+		err = s.ctx.UpdateChannelState(channel, -1, 1)
+	}
+	if err != nil {
+		nsqd.NsqLogger().LogErrorf("failure in %s - %s", req.URL.Path, err)
+		return nil, http_api.Err{500, "INTERNAL_ERROR"}
+	}
+
+	// pro-actively persist metadata so in case of process failure
+	s.ctx.persistMetadata()
```

Comment:
Missing skipped meta data in `persistMetadata` .  The struct `ChannelMetaInfo` should be changed.